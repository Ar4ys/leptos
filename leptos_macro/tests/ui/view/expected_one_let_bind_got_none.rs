use leptos::*;

#[slot]
struct Then<T, C, RC>
where
    C: Fn(T) -> RC,
    RC: IntoView,
{
    data: T,
    children: C,
}

#[component]
fn Slot<T, C, RC>(then: Then<T, C, RC>) -> impl IntoView
where
    C: Fn(T) -> RC,
    RC: IntoView,
{
    _ = then;
}

#[component]
fn Component<T, C, RC>(data: T, children: C) -> impl IntoView
where
    C: Fn(T) -> RC,
    RC: IntoView,
{
    _ = data;
    _ = children;
}

fn main() {
    view! {
        <Component
            data=0
        >
            <p/>
        </Component>
    };

    view! {
        <Slot>
            <Then
                slot
                data=0
            >
                <p/>
            </Then>
        </Slot>
    };
}
