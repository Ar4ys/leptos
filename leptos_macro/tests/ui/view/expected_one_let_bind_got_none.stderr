error[E0283]: type annotations needed for `ComponentProps<i32, C, RC>`
  --> tests/ui/view/expected_one_let_bind_got_none.rs:34:9
   |
34 | /         <Component
35 | |             data=0
36 | |         >
37 | |             <p/>
38 | |         </Component>
   | |____________________^
   |
   = note: multiple `impl`s satisfying `_: Fn(i32)` found in the following crates: `alloc`, `core`:
           - impl<A, F> std::ops::Fn<A> for &F
             where A: Tuple, F: std::ops::Fn<A>, F: ?Sized;
           - impl<Args, F, A> std::ops::Fn<Args> for Box<F, A>
             where Args: Tuple, F: std::ops::Fn<Args>, A: Allocator, F: ?Sized;
note: required by a bound in `Component`
  --> tests/ui/view/expected_one_let_bind_got_none.rs:25:8
   |
23 | fn Component<T, C, RC>(data: T, children: C) -> impl IntoView
   |    --------- required by a bound in this function
24 | where
25 |     C: Fn(T) -> RC,
   |        ^^^^^^^^^^^ required by this bound in `Component`
help: consider giving `props` an explicit type, where the type for type parameter `C` is specified
   |
38 |         </Component>: ComponentProps<i32, C, RC>
   |                     ++++++++++++++++++++++++++++

error[E0283]: type annotations needed for `ComponentProps<i32, C, RC>`
  --> tests/ui/view/expected_one_let_bind_got_none.rs:34:9
   |
34 | /         <Component
35 | |             data=0
36 | |         >
37 | |             <p/>
38 | |         </Component>
   | |____________________^
   |
   = note: multiple `impl`s satisfying `_: Fn(i32)` found in the following crates: `alloc`, `core`:
           - impl<A, F> std::ops::Fn<A> for &F
             where A: Tuple, F: std::ops::Fn<A>, F: ?Sized;
           - impl<Args, F, A> std::ops::Fn<Args> for Box<F, A>
             where Args: Tuple, F: std::ops::Fn<Args>, A: Allocator, F: ?Sized;
note: required by a bound in `ComponentPropsBuilder::<T, C, RC, (__data, ())>::children`
  --> tests/ui/view/expected_one_let_bind_got_none.rs:25:8
   |
23 | fn Component<T, C, RC>(data: T, children: C) -> impl IntoView
   |                                 -------- required by a bound in this associated function
24 | where
25 |     C: Fn(T) -> RC,
   |        ^^^^^^^^^^^ required by this bound in `ComponentPropsBuilder::<T, C, RC, (__data, ())>::children`
help: consider giving `props` an explicit type, where the type for type parameter `C` is specified
   |
38 |         </Component>: ComponentProps<i32, C, RC>
   |                     ++++++++++++++++++++++++++++

error[E0283]: type annotations needed for `ComponentProps<i32, C, RC>`
  --> tests/ui/view/expected_one_let_bind_got_none.rs:34:9
   |
34 | /         <Component
35 | |             data=0
36 | |         >
37 | |             <p/>
38 | |         </Component>
   | |____________________^
   |
   = note: multiple `impl`s satisfying `_: ToChildren<{closure@$DIR/tests/ui/view/expected_one_let_bind_got_none.rs:37:13: 37:17}>` found in the `leptos` crate:
           - impl<F> ToChildren<F> for Box<(dyn FnMut() -> Fragment + 'static)>
             where <F as FnOnce<()>>::Output == Fragment, F: FnMut(), F: 'static;
           - impl<F> ToChildren<F> for Box<(dyn FnOnce() -> Fragment + 'static)>
             where <F as FnOnce<()>>::Output == Fragment, F: FnOnce(), F: 'static;
           - impl<F> ToChildren<F> for Box<(dyn std::ops::Fn() -> Fragment + 'static)>
             where <F as FnOnce<()>>::Output == Fragment, F: Fn(), F: 'static;
           - impl<F> ToChildren<F> for Rc<(dyn std::ops::Fn() -> Fragment + 'static)>
             where <F as FnOnce<()>>::Output == Fragment, F: Fn(), F: 'static;
help: consider giving `props` an explicit type, where the placeholders `_` are specified
   |
38 |         </Component>: ComponentProps<i32, C, RC>
   |                     ++++++++++++++++++++++++++++

error[E0283]: type annotations needed for `ComponentProps<i32, C, RC>`
  --> tests/ui/view/expected_one_let_bind_got_none.rs:34:10
   |
34 |         <Component
   |          ^^^^^^^^^
   |
   = note: multiple `impl`s satisfying `_: Fn(i32)` found in the following crates: `alloc`, `core`:
           - impl<A, F> std::ops::Fn<A> for &F
             where A: Tuple, F: std::ops::Fn<A>, F: ?Sized;
           - impl<Args, F, A> std::ops::Fn<Args> for Box<F, A>
             where Args: Tuple, F: std::ops::Fn<Args>, A: Allocator, F: ?Sized;
note: required by a bound in `ComponentPropsBuilder::<T, C, RC, ((T,), (C,))>::build`
  --> tests/ui/view/expected_one_let_bind_got_none.rs:25:8
   |
22 | #[component]
   | ------------ required by a bound in this associated function
...
25 |     C: Fn(T) -> RC,
   |        ^^^^^^^^^^^ required by this bound in `ComponentPropsBuilder::<T, C, RC, ((T,), (C,))>::build`
help: consider giving `props` an explicit type, where the type for type parameter `C` is specified
   |
38 |         </Component>: ComponentProps<i32, C, RC>
   |                     ++++++++++++++++++++++++++++

error[E0283]: type annotations needed for `Then<i32, C, RC>`
  --> tests/ui/view/expected_one_let_bind_got_none.rs:43:13
   |
42 |           <Slot>
   |            ---- type must be known at this point
43 | /             <Then
44 | |                 slot
45 | |                 data=0
46 | |             >
47 | |                 <p/>
48 | |             </Then>
   | |___________________^
   |
   = note: multiple `impl`s satisfying `_: Fn(i32)` found in the following crates: `alloc`, `core`:
           - impl<A, F> std::ops::Fn<A> for &F
             where A: Tuple, F: std::ops::Fn<A>, F: ?Sized;
           - impl<Args, F, A> std::ops::Fn<Args> for Box<F, A>
             where Args: Tuple, F: std::ops::Fn<Args>, A: Allocator, F: ?Sized;
note: required by a bound in `Slot`
  --> tests/ui/view/expected_one_let_bind_got_none.rs:16:8
   |
14 | fn Slot<T, C, RC>(then: Then<T, C, RC>) -> impl IntoView
   |    ---- required by a bound in this function
15 | where
16 |     C: Fn(T) -> RC,
   |        ^^^^^^^^^^^ required by this bound in `Slot`
help: consider giving `slot` an explicit type, where the type for type parameter `C` is specified
   |
48 |             </Then>: Then<i32, C, RC>
   |                    ++++++++++++++++++

error[E0283]: type annotations needed for `Then<i32, C, RC>`
  --> tests/ui/view/expected_one_let_bind_got_none.rs:43:13
   |
43 | /             <Then
44 | |                 slot
45 | |                 data=0
46 | |             >
47 | |                 <p/>
48 | |             </Then>
   | |___________________^
   |
   = note: multiple `impl`s satisfying `_: Fn(i32)` found in the following crates: `alloc`, `core`:
           - impl<A, F> std::ops::Fn<A> for &F
             where A: Tuple, F: std::ops::Fn<A>, F: ?Sized;
           - impl<Args, F, A> std::ops::Fn<Args> for Box<F, A>
             where Args: Tuple, F: std::ops::Fn<Args>, A: Allocator, F: ?Sized;
note: required by a bound in `SlotPropsBuilder::<T, C, RC>::then`
  --> tests/ui/view/expected_one_let_bind_got_none.rs:16:8
   |
14 | fn Slot<T, C, RC>(then: Then<T, C, RC>) -> impl IntoView
   |                   ---- required by a bound in this associated function
15 | where
16 |     C: Fn(T) -> RC,
   |        ^^^^^^^^^^^ required by this bound in `SlotPropsBuilder::<T, C, RC>::then`
help: consider giving `slot` an explicit type, where the type for type parameter `C` is specified
   |
48 |             </Then>: Then<i32, C, RC>
   |                    ++++++++++++++++++

error[E0283]: type annotations needed for `Then<i32, C, RC>`
  --> tests/ui/view/expected_one_let_bind_got_none.rs:43:13
   |
43 | /             <Then
44 | |                 slot
45 | |                 data=0
46 | |             >
47 | |                 <p/>
48 | |             </Then>
   | |___________________^
   |
   = note: multiple `impl`s satisfying `_: Fn(i32)` found in the following crates: `alloc`, `core`:
           - impl<A, F> std::ops::Fn<A> for &F
             where A: Tuple, F: std::ops::Fn<A>, F: ?Sized;
           - impl<Args, F, A> std::ops::Fn<Args> for Box<F, A>
             where Args: Tuple, F: std::ops::Fn<Args>, A: Allocator, F: ?Sized;
note: required by a bound in `Then`
  --> tests/ui/view/expected_one_let_bind_got_none.rs:6:8
   |
4  | struct Then<T, C, RC>
   |        ---- required by a bound in this struct
5  | where
6  |     C: Fn(T) -> RC,
   |        ^^^^^^^^^^^ required by this bound in `Then`
help: consider giving `slot` an explicit type, where the type for type parameter `C` is specified
   |
48 |             </Then>: Then<i32, C, RC>
   |                    ++++++++++++++++++

error[E0283]: type annotations needed for `Then<i32, C, RC>`
  --> tests/ui/view/expected_one_let_bind_got_none.rs:43:13
   |
43 | /             <Then
44 | |                 slot
45 | |                 data=0
46 | |             >
47 | |                 <p/>
48 | |             </Then>
   | |___________________^
   |
   = note: multiple `impl`s satisfying `_: Fn(i32)` found in the following crates: `alloc`, `core`:
           - impl<A, F> std::ops::Fn<A> for &F
             where A: Tuple, F: std::ops::Fn<A>, F: ?Sized;
           - impl<Args, F, A> std::ops::Fn<Args> for Box<F, A>
             where Args: Tuple, F: std::ops::Fn<Args>, A: Allocator, F: ?Sized;
note: required by a bound in `ThenBuilder::<T, C, RC, (__data, ())>::children`
  --> tests/ui/view/expected_one_let_bind_got_none.rs:6:8
   |
6  |     C: Fn(T) -> RC,
   |        ^^^^^^^^^^^ required by this bound in `ThenBuilder::<T, C, RC, (__data, ())>::children`
...
10 |     children: C,
   |     -------- required by a bound in this associated function
help: consider giving `slot` an explicit type, where the type for type parameter `C` is specified
   |
48 |             </Then>: Then<i32, C, RC>
   |                    ++++++++++++++++++

error[E0283]: type annotations needed for `Then<i32, C, RC>`
  --> tests/ui/view/expected_one_let_bind_got_none.rs:43:13
   |
43 | /             <Then
44 | |                 slot
45 | |                 data=0
46 | |             >
47 | |                 <p/>
48 | |             </Then>
   | |___________________^
   |
   = note: multiple `impl`s satisfying `_: ToChildren<{closure@$DIR/tests/ui/view/expected_one_let_bind_got_none.rs:47:17: 47:21}>` found in the `leptos` crate:
           - impl<F> ToChildren<F> for Box<(dyn FnMut() -> Fragment + 'static)>
             where <F as FnOnce<()>>::Output == Fragment, F: FnMut(), F: 'static;
           - impl<F> ToChildren<F> for Box<(dyn FnOnce() -> Fragment + 'static)>
             where <F as FnOnce<()>>::Output == Fragment, F: FnOnce(), F: 'static;
           - impl<F> ToChildren<F> for Box<(dyn std::ops::Fn() -> Fragment + 'static)>
             where <F as FnOnce<()>>::Output == Fragment, F: Fn(), F: 'static;
           - impl<F> ToChildren<F> for Rc<(dyn std::ops::Fn() -> Fragment + 'static)>
             where <F as FnOnce<()>>::Output == Fragment, F: Fn(), F: 'static;
help: consider giving `slot` an explicit type, where the placeholders `_` are specified
   |
48 |             </Then>: Then<i32, C, RC>
   |                    ++++++++++++++++++

error[E0283]: type annotations needed for `Then<i32, C, RC>`
  --> tests/ui/view/expected_one_let_bind_got_none.rs:43:13
   |
42 |           <Slot>
   |            ---- required by a bound introduced by this call
43 | /             <Then
44 | |                 slot
45 | |                 data=0
46 | |             >
47 | |                 <p/>
48 | |             </Then>
   | |___________________^
   |
   = note: multiple `impl`s satisfying `_: Fn(i32)` found in the following crates: `alloc`, `core`:
           - impl<A, F> std::ops::Fn<A> for &F
             where A: Tuple, F: std::ops::Fn<A>, F: ?Sized;
           - impl<Args, F, A> std::ops::Fn<Args> for Box<F, A>
             where Args: Tuple, F: std::ops::Fn<Args>, A: Allocator, F: ?Sized;
note: required by a bound in `SlotPropsBuilder::<T, C, RC, ((Then<T, C, RC>,),)>::build`
  --> tests/ui/view/expected_one_let_bind_got_none.rs:16:8
   |
13 | #[component]
   | ------------ required by a bound in this associated function
...
16 |     C: Fn(T) -> RC,
   |        ^^^^^^^^^^^ required by this bound in `SlotPropsBuilder::<T, C, RC, ((Then<T, C, RC>,),)>::build`
help: consider giving `slot` an explicit type, where the type for type parameter `C` is specified
   |
48 |             </Then>: Then<i32, C, RC>
   |                    ++++++++++++++++++
